packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -216,6 +220,85 @@ export interface ClusterOptions {
    * @default true
    */
   readonly outputConfigCommand?: boolean;
+
+  /**
+   * Configure access to the Kubernetes API server endpoint.
+   * This feature is only available for kubectl enabled clusters, i.e `kubectlEnabled: true`.
+   *
+   * @see https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html
+   *
+   * @default - Private and Public. The cluster endpoint is accessible from outside of your VPC.
+   * Worker node traffic will leave your VPC to connect to the endpoint
+   */
+  readonly endpointAccess?: EndpointAccess;
+}
+
+/**
+ * Endpoint access characteristics.
+ */
+export class EndpointAccess {
+
+  /**
+   * The cluster endpoint is accessible from outside of your VPC.
+   * Worker node traffic will leave your VPC to connect to the endpoint.
+   *
+   * By default, the endpoint is exposed to all adresses. You can optionally limit the CIDR blocks that can access the public endpoint.
+   * If you limit access to specific CIDR blocks, you must ensure that the CIDR blocks that you
+   * specify include the addresses that worker nodes and Fargate pods (if you use them)
+   * access the public endpoint from.
+   *
+   * @param cidr The CIDR blocks.
+   */
+  public static public(...cidr: string[]) {
+    return new EndpointAccess(false, true, cidr);
```

Comment:
It's not intuitive to me that the positional argument here is a list of cidr blocks. Usually positional arguments should be semantically related to the name of the method. 

This is even more emphasized in the `privateAndPublic` method.

I would either change this to a struct or reconsider this pattern. Perhaps do something like what the backend api looks like:

```ts
endpointAccess: {
  private?: boolean;
  public?: boolean;
  cidr?: string[];
}
```

Reply:
> or reconsider this pattern

You mean reconsider these static methods? I actually really like it, and wouldn't want the user to configure a struct with mutually exclusive properties. Especially not because of how an internal constructor looks like. 

This is modeled after the console experience:

<img width="804" alt="Screen Shot 2020-07-15 at 10 10 01 PM" src="https://user-images.githubusercontent.com/1428812/88534990-d385bc80-d011-11ea-855a-c826cd1140ed.png">

> It's not intuitive to me that the positional argument here is a list of cidr blocks. Usually positional arguments should be semantically related to the name of the method.

I agree, but given I want keep these static methods, What is your suggestion? Instead of the private constructor accepting three arguments it should accept a struct? Not sure I follow.

Reply:
Perhaps something like this:

```ts
endpointAccess: eks.EndpointAccess.PUBLIC,
endpointAccess: eks.EndpointAccess.PRIVATE,
endpointAccess: eks.EndpointAccess.PUBLIC_AND_PRIVATE,
```

And add an `onlyFrom(...cidr)` method:

```ts
endpointAccess: eks.EndpointAccess.PUBLIC.onlyFrom('0.0.0.0/0', ...)
```

I don't love it...

Reply:
I kind of like the `onlyFrom` method...and it does have the benefit of not having to call a static method with no arguments (`EndpointAccess.private()`). But i'm still not loving the full ergonomics either, its a bit less discoverable since it requires two steps to configure CIDR and feels a bit bulky. 

Do you prefer this over the static methods?

Reply:
I think I do. It optimizes for the common use case.

Reply:
Solid point. I'll change.

Reply:
@eladb Made the change. Note that it creates a quirk where `onlyFrom` can be called twice. Since the semantic is **onlyFrom**, I implemented it as overrides, not additive. We can also change the semantic to `allowFrom`, which would make the additive behavior sensible and avoid this quirk. Not sure yet which i like better. 

Also, the `onlyFrom` method is available even when configuring `EndpointAccess.PRIVATE` - in which case it throws. I tried structuring it in a way that doesn't expose this method in that case, but its kind of messy since it requires 2 separate classes for what is essentially the same "enum".